### PR TITLE
Discard phase control if controlled branch is null or open

### PR DESCRIPTION
**Pull request by annetill**
Signed-off-by: Anne Tilloy <anne.tilloy@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

No check is done.

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

Two checks have been added and in case of null controlled branch or open controlled branch, the phase control is not created.

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/network/impl/LfNetworkLoaderImpl.java [hunk 1]
```diff
@@ -249,6 +249,14 @@ private static void createPhaseControl(LfNetwork lfNetwork, PhaseTapChanger ptc,
                 controlledBranchId += legId;
             }
             LfBranch controlledBranch = lfNetwork.getBranchById(controlledBranchId);
+            if (controlledBranch == null) {
+                LOGGER.warn("Phase controlled branch {} is null: no phase control created", controlledBranchId);
+                return;
+            }
+            if (controlledBranch.getBus1() == null || controlledBranch.getBus2() == null) {
```
Comment (floriand-e2r) [inline src/main/java/com/powsybl/openloadflow/network/impl/LfNetworkLoaderImpl.java:256]: Is this check really useful? It seems equivalent to the one at line 265 `ptc.getRegulationTerminal().getBusView().getBus() == null`. In the tests at least there's no counterexample. Thus we never go through the line 266 anymore in the tests.

**Review by floriand-e2r — CHANGES_REQUESTED** [review]

File: src/main/java/com/powsybl/openloadflow/network/impl/LfNetworkLoaderImpl.java (hunk 1 above)
Comment (annetill) [reply src/main/java/com/powsybl/openloadflow/network/impl/LfNetworkLoaderImpl.java:256]: The controlled branch could be open at one side: it could be the side of the regulating terminal or the other one.

Comment (floriand-e2r) [reply src/main/java/com/powsybl/openloadflow/network/impl/LfNetworkLoaderImpl.java:256]: Ok. But then we should remove the test of line 265 which is useless. And it would be maybe better to add such a test of a null branch on the other side? Indeed if I comment out the lines 256-259 all tests stay passed, which shows that when it's null it's always on the regulating side.
